File: 02-week-2--recursion/01-day-1--recursive-counting/solutions/recursive_count.rb
def recursive_count(num = 0)
  # BASE CASE
  return if num >= 10

  puts num
  recursive_count(num + 1)
  # IF YOU PUTS NUM BELOW, THE NUMBERS PRINT BACKWARDS FROM 9 TO 0. WHY?
  # puts num
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

File: 02-week-2--recursion/01-day-1--recursive-counting/solutions/recursive_count.js
function recursiveCount(num = 0) {
  if (num >= 10) {
    return;
  }

  console.log(num);
  recursiveCount(num + 1);

  // THIS IS ALSO A VALID OPTION INSTEAD OF THE ABOVE
  // recursiveCount(++num);

  // THIS WILL CAUSE A STACK OVERFLOW. WHY?
  // recursiveCount(num++);

  // IF YOU LOG NUM BELOW, THE NUMBERS PRINT BACKWARDS FROM 9 TO 0. WHY?
  // console.log(num);
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
